support/Python/DirectoryStructure.py

```diff
@@ -126,3 +126,99 @@ def list_segments(segments_dir: Union[str, Path]) -> List[Segment]:
         return []
     matches = map(Segment.match, segments_dir.iterdir())
     return sorted(match for match in matches if match)
+
+
+@dataclass(frozen=True, order=True)
+class PipelineStep:
+    """A complete simulation iteration
+
+    Represents a full simulation run. Multiple iterations can exist,
+    organized in a directory structure like this:
+
+    ```
+    BASE_DIR/
+        001_InitialData/
+            Segments/
+                Segment_0000/
+                    Checkpoints/
+                    InputFile.yaml
+                    Submit.sh
+                Segment_0001/
+                    ...
+        002_Inspiral/
```

Comment:
Evolutions use segments, so list `Segment_0000/` and `Segment_0001/` here (note that there's no enclosing `Segments/` directory)